OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnAttack.cs

```diff
@@ -0,0 +1,172 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class GrantConditionOnAttackInfo : ITraitInfo
+	{
+		[GrantedConditionReference]
+		[Desc("The condition type to grant.")]
+		public readonly string Condition = null;
+
+		[WeaponReference]
+		[Desc("Name of the armament that grants this condition.")]
+		public readonly string ArmamentName = null;
+
+		[Desc("Shots required to apply an instance of the condition. If there are more instances of the condition granted than values listed,",
+			"the last value is used for all following instances beyond the defined range.")]
+		public readonly int[] RequiredShotsPerInstance = { 3 };
```

Comment:
This default value looks a bit arbitrary, I'd prefer a default of 1 because the main (or even only) usage in the official mods will likely be to grant a single-instance condition with the first shot.